ngraph/frontend/onnx_editor/include/onnx_editor/edge_mapper.hpp

```diff
@@ -0,0 +1,95 @@
+
+#pragma once
+
+#include <map>
+#include <string>
+#include <vector>
+
+#include "onnx_editor/editor_types.hpp"
+
+namespace ONNX_NAMESPACE
+{
+    // forward declaration to avoid the necessity of include paths setting in components
+    // that don't directly depend on the ONNX library
+    class GraphProto;
+} // namespace ONNX_NAMESPACE
+
+namespace ngraph
+{
+    namespace onnx_editor
+    {
+        /// \brief A class allows to determine InputEdge and OutputEdge by user-friendly onnx names.
+        class EdgeMapper
+        {
+        public:
+            EdgeMapper() = delete;
+
+            /// \brief Creates an edge mapper based on graph_proto state.
+            ///
+            /// \note If state of graph_proto will be changed, the information
+            ///       from edge mapper is outdated.
+            ///
+            /// \param graph_proto Reference to GraphProto object.
+            EdgeMapper(const ONNX_NAMESPACE::GraphProto& graph_proto);
+
+            /// \brief Returns the InputEdge based on a node (node name or output name)
+            ///        and an input (input name or input index).
+            ///
+            /// \note  The node name can be ambiguous (many nodes can have the same name).
+            ///        In such a case the algorthim try to match the given node name
+            ///        with the input name (given input index is not enough).
+            ///        If after such operation a found edge is unique, it is returned.
+            ///        If InputEdge cannot be determined based on given params the ngraph_error
+            ///        exception is thrown.
+            ///
+            /// \param node A node helper structure created based on a node name
+            ///             or a node output name.
+            ///
+            /// \param input A input helper structure created based on a input name
+            ///              or a input index.
+            InputEdge to_input_edge(const Node& node, const Input& input) const;
+
+            /// \brief Returns the OutputEdge based on a node (node name or output name)
+            ///        and an output (output name or output index).
+            ///
+            /// \note  The node name can be ambiguous (many nodes can have the same name).
+            ///        In such a case the algorthim try to match the given node name
+            ///        with the output name (given output index is not enough).
+            ///        If after such operation a found edge is unique, it is returned.
+            ///        If OutputEdge cannot be determined based on given params the ngraph_error
+            ///        exception is thrown.
+            ///
+            /// \param node A node helper structure created based on a node name
+            ///             or a node output name.
+            ///
+            /// \param output A output helper structure created based on a output name
+            ///               or a output index.
```

Comment:
This comment is repeating a lot of things from `to_input_edge`. Please shorten and make a reference to look up more information in the other function.

```suggestion
            /// \brief Returns an OutputEdge based on a node (node name or output name)
            ///        and an output (output name or output index).
            ///
            /// \note  The node name can be ambiguous (many ONNX nodes can have the same name).
            ///        In such a case the algorthim will try to match the given node name
            ///        with the output name (providing an output index is not enough).
            ///        If after such operation a found edge is unique, it is returned.
            ///        If OutputEdge cannot be determined based on given params the ngraph_error
            ///        exception is thrown.
            ///
            /// \param node A node helper structure created based on a node name
            ///             or a node output name.
            ///
            /// \param output A output helper structure created based on a output name
            ///               or a output index.
```